lib/ntlmauth/ntlmauth.h

```diff
@@ -32,24 +32,25 @@ extern "C" {
 #define NTLM_REQUEST_NON_NT_SESSION_KEY     0x400000
 
 /* NTLM error codes */
-#define NTLM_ERR_INTERNAL         -3
-#define NTLM_ERR_BLOB             -2
-#define NTLM_ERR_BAD_PROTOCOL     -1
-#define NTLM_ERR_NONE              0    /* aka. SMBLM_ERR_NONE */
-/* codes used by smb_lm helper */
-#define NTLM_ERR_SERVER            1    /* aka. SMBLM_ERR_SERVER   */
-#define NTLM_ERR_PROTOCOL          2    /* aka. SMBLM_ERR_PROTOCOL */
-#define NTLM_ERR_LOGON             3    /* aka. SMBLM_ERR_LOGON    */
-#define NTLM_ERR_UNTRUSTED_DOMAIN  4
-#define NTLM_ERR_NOT_CONNECTED     10
-/* codes used by mswin_ntlmsspi helper */
-#define NTLM_SSPI_ERROR         1
-#define NTLM_BAD_NTGROUP        2
-#define NTLM_BAD_REQUEST        3
-/* TODO: reduce the above codes down to one set non-overlapping. */
+enum class NtlmError : int
+{
+    None = 0,
+    ServerError = 1,
+    ProtocolError = 2,
+    LoginEror = 3,
+    UntrustedDomain = 4,
+    NtlmNotConnected = 5,
+    SspiError = 6,
+    BadNtGroup = 7,
+    BadRequest = 8,
+    InternalError = 128,
+    BlobError = 129,
+    BadProtocol = 130
```

Comment:
The `None` value has to be zero because the current code relies on that (and because it is good practice). After your [clarification](https://github.com/squid-cache/squid/pull/676#discussion_r443688039), I see no need (and see harm in) explicitly numbering all other values, especially numbering them with a strange undocumented gap  after value `8`. Please remove explicit values other than zero.

Reply:
ok